src/connector/src/source/kafka/private_link.rs

```diff
@@ -166,3 +175,141 @@ pub fn insert_privatelink_broker_rewrite_map(
     with_options.insert(PRIVATE_LINK_BROKER_REWRITE_MAP_KEY.to_owned(), json);
     Ok(())
 }
+
+fn handle_privatelink_endpoint(
+    endpoint: &str,
+    broker_rewrite_map: &mut HashMap<String, String>,
+    link_targets: &[AwsPrivateLinkItem],
+    broker_addrs: &[&str],
+) -> ConnectorResult<()> {
+    let endpoint = if let Ok(json) = serde_json::from_str::<serde_json::Value>(endpoint) {
```

Comment:
here we introduce a potential risk is if the user input an illegal json, which cannot be parsed, we will see it as a string, instead of returning error. The case seems acceptable to me because he cannot connect to the broker using the wrong endpoint addr and he will find out later.